src/molecule/provisioner/ansible_playbook.py

```diff
@@ -53,6 +53,24 @@ def __init__(self, playbook, config, verify=False):
         else:
             self._env = self._config.provisioner.env
 
+        # merge paths with runtime
+        roles_paths = util.merge_paths(
+            self._env.get("ANSIBLE_ROLES_PATH"),
+            self._config.runtime.environ.get("ANSIBLE_ROLES_PATH"),
+        )
+        library_paths = util.merge_paths(
+            self._env.get("ANSIBLE_LIBRARY"),
+            self._config.runtime.environ.get("ANSIBLE_LIBRARY"),
+        )
+        collections_paths = util.merge_paths(
+            self._env.get(config.ansible_collections_path),
+            self._config.runtime.environ.get(config.ansible_collections_path),
+        )
+
+        self._env["ANSIBLE_ROLES_PATH"] = roles_paths
+        self._env["ANSIBLE_LIBRARY"] = library_paths
+        self._env[config.ansible_collections_path] = collections_paths
+
```

Comment:
If I understand it correctly, you want to append `cache_dir` created by `ansible-compat` in `ANSIBLE_ROLES_PATH`.

It has been inserted into `ANSIBLE_ROLES_PATH` at https://github.com/ansible-community/molecule/blob/d263e982dec4379dbd93a3f2addd6c5501b1e4ad/src/molecule/provisioner/ansible.py#L446-L468

Is it duplicate?

Reply:
Correct! This was fixed with this commit by Jack as of yesterday: https://github.com/ansible-community/molecule/commit/b9d8f7f30560acbe30cb4705e69d5b264d9ecca9

Since his got merged first, I'll remove the code changes I made, but keep the tests.